x-pack/plugins/reporting/server/browsers/chromium/driver_factory/index.js

```diff
@@ -49,6 +49,37 @@ export class HeadlessChromiumDriverFactory {
         });
 
         page = await chromium.newPage();
+        await page.setRequestInterception(true);
+
+        page.on('request', req => {
+          this.logger.debug(`Request: [${req.url()}]`);
+          req.continue();
+        });
+        page.on('response', res => {
+          this.logger.debug(`Response: [${res.url()} - ${res.status()}]`); // TODO: test more this logs more than just 200 responses
+        });
+        page.on('requestfailed', async failedReq => {
+          const { errorText } = failedReq.failure();
+          this.logger.error(
+            `Request failed! [${failedReq.url()}] [${errorText}]`
+          );
+          const failedResponse = failedReq.response();
+          if (failedResponse != null) {
+            const json = await failedResponse.json();
+            this.logger.debug(`Partial response for failed request: ${json}`);
+          } else {
+            this.logger.debug(`No partial response for failed request`);
+          }
+        });
+        page.on('console', msg => {
+          const args = msg.args();
```

Comment:
Yeah, I kind of copypasta'd the sample code on the puppeteer API for this. 

It might make more sense to actually keep it this way since the messages getting logged can include a stack trace. If we comma-join it, it'd just be a jumble.